src/main/java/com/vaccin/vaccin/controller/VaccineAppointmentController.java

```diff
@@ -20,15 +21,34 @@ public VaccineAppointmentController( VaccineAppointmentService vaccineAppointmen
     }
 
     @PostMapping("/appointments/add")
-    public String addAppointment(@RequestBody VaccineAppointmentCreateDto vaccineAppointmentCreateDto) {
-
-        return vaccineAppointmentService.appointUser(vaccineAppointmentCreateDto);
-
+    public ResponseEntity<List<VaccineAppointmentDto>> addAppointment
+            (@RequestBody VaccineAppointmentCreateDto vaccineAppointmentCreateDto) {
+
+        try {
+            List<VaccineAppointmentDto> vaccineAppointmentDtoList =
+                vaccineAppointmentService.appointUser(vaccineAppointmentCreateDto);
+            return new ResponseEntity<>(vaccineAppointmentDtoList, HttpStatus.CREATED);
+        } catch (Exception exception) {
+            if (exception.getMessage().equals("User not found")
+                    || exception.getMessage().equals("Vaccine Center not found")) {
+                return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+            } else {
+                return new ResponseEntity<>(HttpStatus.FORBIDDEN);
+            }
+        }
     }
 
     @GetMapping("/appointments/{patientId}")
-    public List<VaccineAppointmentDto> getAppointments(@PathVariable long patientId) {
-
-        return vaccineAppointmentService.getAppointments(patientId);
+    public ResponseEntity<List<VaccineAppointmentDto>> getAppointments(@PathVariable long patientId) {
+
+        try {
+            List<VaccineAppointmentDto> appointments = vaccineAppointmentService.getAppointments(patientId);
+            if (appointments.isEmpty()) {
+                return new ResponseEntity<>(HttpStatus.NO_CONTENT);
```

Comment:
We should discuss with the frontend on whether an empty list should be sent as is or handled differently